test/fuzzyDateTime.js

```diff
@@ -0,0 +1,249 @@
+				for (const delta of deltas) {
+					totalDelta += delta;
+					formatFuzzyDate(d, {
+						origin: new Date(origin.getTime() + totalDelta),
+						onUpdate
+					});
+					await sleep(10);
+				}
+				formatFuzzyDate(d, {
+					origin: new Date(origin.getTime() + 4 * day),
+					onUpdate,
+				});
+
+				deltas.push(1 * minute);
+				deltas.push(-1);
+				expect(actualDeltas).to.deep.equal(deltas.slice(1));
+				expect(invocations).to.deep.equal([
+					'1 second ago',
+					'2 seconds ago',
+					'4 seconds ago',
+					'8 seconds ago',
+					'16 seconds ago',
+					'1 minute ago',
+					'2 minutes ago',
+					'3 minutes ago',
+					'9/23/2015',
+				]);
+			});
+		});
+
+		it('uses absolute formatting for recent times when rtf is not available', () => {
+			expect(formatFuzzyDate(past.seconds(5), { origin: noon })).to.not.match(shortDateRegex);
+			expect(formatFuzzyDate(past.seconds(5), { origin: noon, rtf: false })).to.match(shortDateRegex);
+		});
+
+		relativeExamples.forEach(([makeTime, expected,, { rtf, origin = noon } = {}]) => {
+			it(`formats ${label(makeTime)} as ${expected}`, () => {
+				expect(formatFuzzyDateTime(makeTime(), { origin, rtf })).to.equal(expected);
+			});
+		});
+
+		it('formats > 3 days as absolute', () => {
+			expect(formatFuzzyDateTime(past.days(3.5))).to.match(shortDateTimeRegex);
+			expect(formatFuzzyDateTime(future.days(3.5))).to.match(shortDateTimeRegex);
+		});
+
+		const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+		for (const [makeTime,, expected, { origin = noon } = {}] of relativeExamples) {
+			it(`updates ${label(makeTime)} in ${label(expected)}`, (done) => {
+				formatFuzzyDateTime(makeTime(), {
+					origin,
+					onUpdate: (text, opt) => {
+						expect(opt.nextUpdateInMilliseconds)
+							.to.be.closeTo(expected(), 100);
+						done();
+						return { stopUpdates: true };
+					}
+				});
+			});
+		}
+
+		it('optionally provides updates via a callback', (done) => {
+			let callCount = 0;
+			const expectedText = ['1 second ago', '2 seconds ago'];
+			const expectedTimeout = [1000, 2000];
+			const immediateResult = formatFuzzyDateTime(past.seconds(1, new Date()), {
+				onUpdate: (text, opt = {}) => {
+					expect(text).to.equal(expectedText[callCount]);
+					expect(opt.nextUpdateInMilliseconds).to.be.closeTo(expectedTimeout[callCount], 100);
+					callCount += 1;
+					if (callCount === expectedText.length) {
+						done();
+						return { stopUpdates: true };
+					}
+				}
+			});
+			expect(immediateResult).to.equal(expectedText[0]);
+		}).timeout(3000);
+
+		it('the next update can be cancelled', (done) => {
+			let callCount = 0;
+			const immediateResult = formatFuzzyDateTime(past.seconds(1, new Date()), {
+				onUpdate: () => {
+					callCount += 1;
+					return { stopUpdates: true };
+				}
+			});
+			expect(immediateResult).to.equal('1 second ago');
+			sleep(1500).then(() => {
+				expect(callCount).to.equal(1);
+				done();
+			});
+		}).timeout(3000);
```

Comment:
To speed up these tests, I'd suggest using Sinon's fake timers. They'll let you fast forward time without actually delaying.